src/handle_query_contract_ui.c

```diff
@@ -20,7 +20,20 @@ static void set_payable_amount_ui(ethQueryContractUI_t *msg, context_t *context)
 static void set_amount_ui(ethQueryContractUI_t *msg, context_t *context) {
     strlcpy(msg->title, "Quantity", msg->titleLength);
 
-    amountToString(context->amount, sizeof(context->amount), 0, "", msg->msg, msg->msgLength);
+    if (context->selectorIndex == MINT_SIGN_V2) {
+        // Convert the amount to a uint32_t
+        uint32_t amountInt = U4BE(context->amount, PARAMETER_LENGTH - sizeof(amountInt));
```

Comment:
When extracting a `uint16_t/uint32_t` value from a parameter these functions should be used https://github.com/LedgerHQ/ethereum-plugin-sdk/blob/b9777e7a81a9c33b3f8997ffa23396fa2b3f861d/include/eth_internals.c#L263-L272